jdaviz/configs/cubeviz/plugins/parsers.py

```diff
@@ -147,63 +285,51 @@ def _parse_jwst_s3d(app, hdulist, data_label, ext='SCI', viewer_name='flux-viewe
                     hdulist[ext].header['MJD-OBS'] = t.mjd
                     break
 
-    if ext == 'DQ':  # DQ flags have no unit
-        flux = hdulist[ext].data << u.dimensionless_unscaled
-    else:
-        unit = u.Unit(hdulist[ext].header.get('BUNIT', 'count'))
-        flux = hdulist[ext].data << unit
-    wcs = WCS(hdulist['SCI'].header, hdulist)  # Everything uses SCI WCS
-    data = Spectrum1D(flux, wcs=wcs)
-
-    # NOTE: Tried to only pass in sliced WCS but got error in Glue.
-    # sliced_wcs = wcs[:, 0, 0]  # Only want wavelengths
-    # data = Spectrum1D(flux, wcs=sliced_wcs)
-
-    app.add_data(data, data_label)
-    app.add_data_to_viewer(viewer_name, data_label)
-    if viewer_name == 'flux-viewer':
-        app.add_data_to_viewer('spectrum-viewer', data_label)
-
 
-def _parse_spectrum1d_3d(app, file_obj):
-    # Load spectrum1d as a cube
+def _parse_spectrum1d_3d(app, file_obj, base_data_label):
+    """Load Spectrum1D as a cube."""
 
-    for attr in ["flux", "mask", "uncertainty"]:
+    for attr in ("flux", "uncertainty", "mask"):
         val = getattr(file_obj, attr)
-        if val is None:
+        if val is None:  # pragma: no cover
             continue
 
-        if attr == "mask":
-            flux = val << file_obj.flux.unit
-        elif attr == "uncertainty":
-            if hasattr(val, "array"):
-                flux = u.Quantity(val.array, file_obj.flux.unit)
-            else:
-                continue
-        else:
+        if attr == "flux":
+            data_type = 'flux'
             flux = val
-
-        flux = np.moveaxis(flux, 1, 0)
```

Comment:
Really not sure what this `moveaxis` is for. All it did was to make the output X/Y flipped w.r.t. input. I removed it but if there is a reason for it to be there, please let me know.

Reply:
I think the reason for this had something to do with preserving the orientation that `specutils` understands and expects, or handling a difference between `specutils` and `spectral-cube`?  There was also a related issue of whether or not the input cubes needed to be transposed inside the relevant `specutils` cube loaders.  I can't quite remember.  There was some discussion about it here, https://github.com/astropy/specutils/issues/852, and maybe in some other jdaviz PRs/issues.  @rosteen might remember more specific details.

Reply:
But this line doesn't change where the WCS axis go, just flipping spatial, which is very confusing for me. Maybe I am missing something here.

Reply:
I don't remember if this was originally put in for WCS reasons, or something else.  There might be unintended consequences by removing it.  Maybe someone else can comment on it.  Looks like it was put in by @javerbukh during the spectralcube to specutils transition.

Reply:
I think @javerbukh said that we no longer needs `movaxis`.